pkg/controller/crdmirroring/crdhandler/clustergroup.go

```diff
@@ -0,0 +1,147 @@
+	l := obj.(*legacycore.ClusterGroup)
+	n := c.buildNewObject(l)
+	client := c.client.CrdV1alpha2().ClusterGroups()
+	_, err := client.Create(context.TODO(), n, metav1.CreateOptions{})
+	return err
+}
+
+// SyncObject updates the mirroring new ClusterGroup
+func (c *ClusterGroupHandler) SyncObject(legacyObj, newObj metav1.Object) error {
+	if !c.deepEqualSpecAndLabels(legacyObj, newObj) {
+		n := c.syncNewObject(legacyObj, newObj)
+		newClient := c.client.CrdV1alpha2().ClusterGroups()
+		_, err := newClient.Update(context.TODO(), n, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	if !c.deepEqualStatus(legacyObj, newObj) {
+		l := c.syncLegacyObject(legacyObj, newObj)
+		legacyClient := c.legacyClient.CoreV1alpha2().ClusterGroups()
+		_, err := legacyClient.UpdateStatus(context.TODO(), l, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+//DeleteNewObject deletes the mirroring new ClusterGroup
+func (c *ClusterGroupHandler) DeleteNewObject(namespace, name string) error {
+	client := c.client.CrdV1alpha2().ClusterGroups()
+	return client.Delete(context.TODO(), name, metav1.DeleteOptions{})
+}
+
+// StaleLegacyObject updates the mirroring new ClusterGroup
+func (c *ClusterGroupHandler) StaleLegacyObject(newObj metav1.Object) error {
+	n := newObj.(*crd.ClusterGroup).DeepCopy()
+	delete(n.Annotations, types.ManagedBy)
+	newClient := c.client.CrdV1alpha2().ClusterGroups()
+	_, err := newClient.Update(context.TODO(), n, metav1.UpdateOptions{})
+	return err
+}
+
+// GetLegacyObject gets the legacy ClusterGroup struct
+func (c *ClusterGroupHandler) GetLegacyObject(namespace, name string) (metav1.Object, error) {
+	return c.legacyLister.Get(name)
+}
+
+// buildNewObject returns a new ClusterGroup struct
+func (c *ClusterGroupHandler) buildNewObject(obj metav1.Object) *crd.ClusterGroup {
+	l := obj.(*legacycore.ClusterGroup)
+	n := &crd.ClusterGroup{}
+	n.Spec = *l.Spec.DeepCopy()
+	setMetaData(l, n)
+	return n
+}
+
+// syncNewObject syncs legacy ClusterGroup' Spec and Labels to the new ClusterGroup
+func (c *ClusterGroupHandler) syncNewObject(legacyObj, newObj metav1.Object) *crd.ClusterGroup {
+	l := legacyObj.(*legacycore.ClusterGroup)
+	n := newObj.(*crd.ClusterGroup).DeepCopy()
+	n.Spec = *l.Spec.DeepCopy()
+	n.Labels = labelsDeepCopy(l)
+	return n
+}
+
+// syncLegacyObject syncs new ClusterGroup' Status to the legacy ClusterGroup
+func (c *ClusterGroupHandler) syncLegacyObject(legacyObj, newObj metav1.Object) *legacycore.ClusterGroup {
+	l := legacyObj.(*legacycore.ClusterGroup).DeepCopy()
+	n := newObj.(*crd.ClusterGroup)
+	l.Status = *n.Status.DeepCopy()
+	return l
+}
+
+func (c *ClusterGroupHandler) deepEqualSpecAndLabels(legacyObj, newObj metav1.Object) bool {
+	l := legacyObj.(*legacycore.ClusterGroup)
+	n := newObj.(*crd.ClusterGroup)
+	spec := reflect.DeepEqual(l.Spec, n.Spec)
+	labels := reflect.DeepEqual(l.Labels, n.Labels)
```

Comment:
```
if !reflect.DeepEqual(l.Spec, n.Spec) {
    return false
}
if !reflect.DeepEqual(l.Labels, n.Labels) {
    return false
}
return true
```
This would be more efficient

Reply:
Make sense.